README
SVG-Calendar version 0.1.2
===========================

This module is used to create calendars in standards compliant format.
Currently only monthly pages or whole years ie 12 individual months can be
created. You can create calendars with the supplied svgcal.pl program see:

$ svgcal.pl --help
or
$ svgcal.pl --man

For more information.

INSTALLATION

To install this module type the following:

   perl Build.PL
   ./Build
   ./Build test
   ./Build install

DEPENDENCIES

This module requires these other modules and libraries:

  SVG
  Math::Trig
  Template
  Template::Provider::FromDATA
  Image::ExifTool
  Class::Date
  Clone
  Readonly
  version

COPYRIGHT AND LICENCE

Copyright (c) 2006-2009 Ivan Wills (14 Mullion Close, Hornsby Heights, NSW Australia 2077)
All rights reserved.

This module is free software; you can redistribute it and/or modify it under
the same terms as Perl itself. See L<perlartistic>.  This program is
distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
PARTICULAR PURPOSE.


Changes
Revision history for Perl extension SVG::Calendar

0.3.10   2015-12-05
        Updating for release of version 0.3.10
        Added a little more detail to synopsis (Ivan Wills)

0.3.9   2015-09-02
        Updating for release of version 0.3.9
        Removed creation of makefile (Ivan Wills)

0.3.8   2015-06-24
        Updating for release of version 0.3.8
        Removed unneeded tag (Ivan Wills)
        Added specific Test::More version for the features being used (Ivan Wills)

0.3.7   2015-01-02
        Updating for release of version 0.3.7
        Fixed MANIFEST files to templates are deployed (Ivan Wills)

0.3.6   2015-01-02
        Updating for release of version 0.3.6
        Removed unused module (Ivan Wills)
        Dates updated (Ivan Wills)
        Fixed repeating "class" (Ivan Wills)
        Moved moon inline styling to classes (Ivan Wills)
        Removed dependency on SVG.pm (Ivan Wills)
        Cleaned up white space around tags (Ivan Wills)
        Cleaned up adding empty styles (Ivan Wills)
        Getting rid of SVG.pm reliance (Ivan Wills)
        Updated dependencies (Ivan Wills)
        changed --show-template to use new template (Ivan Wills)
        Started using the share dir path for templates (Ivan Wills)
        Moved template to own file (Ivan Wills)
        Added new TODO file (Ivan Wills)

0.3.5   2014-12-24
        Updating for release of version 0.3.5
        Things to do (Ivan Wills)
        Fixed date checking (Ivan Wills)

0.3.4   2014-12-23
        Updating for release of version 0.3.4
        Updated dependencies (Ivan Wills)
        Added new dependency (Ivan Wills)
        Updated license (Ivan Wills)
        Updated manifest (Ivan Wills)
        Added warning about not specifying any dates (Ivan Wills)
        Fixed moon at night not being a duration object (Ivan Wills)
        Updated tests (Ivan Wills)

0.3.3   2014-08-21
        Updated manifest (Ivan Wills)
        Cleaned up the tests (Ivan Wills)
        Made executable (Ivan Wills)

v0.3.2   2014-06-02
    Fixed executable bit (Ivan Wills)
    Adding tag files (Ivan Wills)
    Added configure requires section (Ivan Wills)
    Added new releases change log (Ivan Wills)

v0.3.1 2012-07-21
    Build updates
    Upped version number
    Fixed up Build.PL script
    Got test to my latest standards
    Updated version to 0.3, added version changes to change log

v0.3.0 2011-12-09
    Updated license text
    Added missing new generated META.json file
    Fixed spelling mistakes, updated list of stop words
    Trying to make t/moon.t more stable verses moon phase calculations
    Added missing requirement
    Removed generated files from repo, updated ignores/skips
    Removed some vestages of Class::Date
    Added recommends section for moon phase calculation modules
    Removed DateTime::Util::Astro::Moon use as it no longer appears to be on CPAN
    More ignore sorting
    Changed from requiring Class::Date to DateTime
    Removed DateTime::Util::Astro::Moon option which doesn't seem to exist any more
    Converted tabs to spaces
    Started to implement detailed config with comments for default values
    implemented changing the moons radius, fixed an issue when no start month supplied and started to implement export options
    Added ignore file
    Cleaned template
    Added some extra logic checking
    Added specifying images by month number, fixed POD

0.2.0 2010-10-02
    Fixed spelling test from failing when no spell command exists
    other code cleanups

0.1.3
    Fixed perl version requirements
    Added perl critic test
    Changed kwalitee test to be author only

0.1.2
    Fixed missing version required module in Meta.yml

0.1.1
    - Added year display to calendar

0.1.0
    - Fixed not showing images
    - Changed the full moon to black (unless it is the second in a month)
    - Fixed scaling of images.
    - Added missing params to svgcal.pl
    - Changed Test::Kwalitee to run if possible

0.0.6 Sat June 7 20:17:00 2008
    - Added missing required module Readonly to list of required modules.

0.0.5 Thu June 5 11:21:00 2008
    - Some documentation updates
    - Fixes for perlcritic and kwalitee

0.01  Fri May  30 14:11:54 2008
    - original version


Makefile.PL
# Note: this file was auto-generated by Module::Build::Compat version 0.3607
require 5.008;
use ExtUtils::MakeMaker;
WriteMakefile
(
          'NAME' => 'SVG::Calendar',
          'VERSION_FROM' => 'lib/SVG/Calendar.pm',
          'PREREQ_PM' => {
                           'Class::Date' => 0,
                           'Clone' => '0.17',
                           'Image::ExifTool' => '7.0',
                           'Math::Trig' => '1.15',
                           'Readonly' => '1.03',
                           'SVG' => '2.33',
                           'Template' => '2.14',
                           'Template::Provider::FromDATA' => '0.08',
                           'Test::Kwalitee' => 0,
                           'Test::More' => 0,
                           'Test::Perl::Critic' => 0,
                           'Test::Pod' => 0,
                           'Test::Pod::Coverage' => 0,
                           'Test::Spelling' => 0,
                           'version' => '0.74'
                         },
          'INSTALLDIRS' => 'site',
          'EXE_FILES' => [
                           'bin/svgcal.pl'
                         ],
          'PL_FILES' => {}
        )
;


Build.PL
use strict;
use warnings;
use Module::Build;

my $builder = Module::Build->new(
    module_name       => 'SVG::Calendar',
    license           => 'perl',
    dist_author       => 'Ivan Wills <ivan.wills@gmail.com>',
    dist_version_from => 'lib/SVG/Calendar.pm',
    requires          => {
        'perl'                       => '5.006',
        'Clone'                      => '0.17',
        'Config::Std'                => 0,
        'DateTime::Format::Strptime' => 0,
        'File::ShareDir'             => '1.102',
        'Image::ExifTool'            => '7.0',
        'Math::Trig'                 => '1.15',
        'Path::Tiny'                 => '0.061',
        'Readonly'                   => '1.03',
        'Scalar::Util'               => 0,
        'Template'                   => '2.14',
        'version'                    => '0.74',
    },
    configure_requires => {
        'Module::Build' => 0.38,
    },
    build_requires => {
        'Test::More'     => '0.88',
        'Test::Warnings' => 0,
    },
    recommends => {
        'Astro::Coord::ECI::Moon' => 0.044,
        'Astro::MoonPhase'        => 0.60,
        'Test::Spelling'          => 0,
        'Test::Perl::Critic'      => 0,
        'Test::Kwalitee'          => 0,
        'Pod::Coverage'           => 0,
    },
    share_dir => {
        dist => [
            'templates',
        ],
    },
    add_to_cleanup     => [ 'SVG-Calendar-*' ],
    create_makefile_pl => 'traditional',
    create_license     => 1,
    meta_merge         => {
        resources => {
            repository => 'git://github.com/ivanwills/SVG-Calendar.git',
            bugtracker => 'http://rt.cpan.org/NoAuth/Bugs.html?Dist=SVG-Calendar',
        },
    },
);

$builder->create_build_script();
